File: app/src/test/java/org/simon/npo/cucumber/step/CommonSteps.java
package org.simon.npo.cucumber.step;

import io.cucumber.java.en.Then;
import org.junit.jupiter.api.Assertions;
import org.simon.npo.cucumber.AbstractStepsDefinitions;
import org.springframework.http.HttpStatusCode;
import org.springframework.http.ResponseEntity;

public class CommonSteps extends AbstractStepsDefinitions {

  @Then("тогда получен код ответа {int}")
  public void startIndirectActivity(Integer responseCode) {
    final ResponseEntity apiResponse = (ResponseEntity) getContext().getApiResponse();
    Assertions.assertEquals(HttpStatusCode.valueOf(responseCode), apiResponse.getStatusCode());
  }
}

File: app/src/test/kotlin/org/simon/npo/cucumber/step/CommonSteps.kt
package org.simon.npo.cucumber.step

import io.cucumber.java.en.Given
import org.simon.npo.core.service.AppDateTimeProvider
import org.simon.npo.cucumber.AbstractStepsDefinitions
import java.time.Clock
import java.time.LocalDateTime
import java.time.ZoneOffset

class CommonSteps(private val appDateTimeProvider: AppDateTimeProvider) : AbstractStepsDefinitions() {
    @Given("сейчас {cucumberDate}")
    fun givenNow(dateTime: LocalDateTime): Unit =
        run {
            Clock.fixed(dateTime.toInstant(ZoneOffset.UTC), ZoneOffset.UTC)
                .also { appDateTimeProvider.clock = it }
        }
}
